src/phpgrep/matcher_test.go

```diff
@@ -422,6 +422,8 @@ func TestMatch(t *testing.T) {
 		{`$x->ff(1, 2)`, `$this->ff(1, 2)`},
 
 		{`$_[0]`, `$v[0]`},
+		{`$_{0}`, `$v{0}`},
+		{`$_{0}`, `$v[0]{0}`},
```

Comment:
You also need to add negative tests.
For example, `$_[$_]` should not match `$_{$_}` and vice versa.